asset.go

```diff
@@ -14,6 +14,16 @@ type Asset struct {
 	Signal signal.Floating
 }
 
+// NewAsset creates an Asset with a preallocated signal buffer
+func NewAsset(channels, capacity int) *Asset {
```

Comment:
Thinking about the generic handling I'd say what you had previously was more idiomatic go. I think having the exported aren't actually clear if they are required / what they do. I see in the tests that Length is also used when creating a new signal.SignalAllocator for example.

Reply:
I think in that case length is required because WriteStripedFloat64 function is used. There is a difference in how `WriteT`/`WriteStripedT` and `Append` functions work: `Write` is using buffers length to calculate number of values, like other `Write` functions from standard library. `Append`, on the opposite, uses capacity to grow buffer when possible. This is also aligned with standard library. 

For the asset I see the following improvements: 

1. Remove Source method to reduce the scope of it;
2. Make SampleRate unexported - it will be provided by pipe via SinkFuncAllocator; new getter can be provided;

It'll leave asset with only exported `signal.Floating` field and `signal.Sink` which will make it pretty obvious how to use it. What do you think?

Reply:
Yeah I think that makes things a lot simpler and thus clearer. Maybe the naming of the exported fields could be a little better to make it clearer what they do?

Reply:
So for the asset having source and sink. I think not having source for asset makes more sense now with the above discussed changes.

I guess some of the confusion can also be attributed to the positioning of the new generic source function which IMO should be in a separate file as it's no longer anything to do with asset itself.